python/mdvtools/dbutils/file_watcher.py

```
@@ -0,0 +1,30 @@
+import os
+
+class ProjectDirectoryHandler(FileSystemEventHandler):
+    def __init__(self, app, watch_path):
+        self.app = app
+        self.watch_path = watch_path
+
+    def on_created(self, event):
+        if event.is_directory:
+            print(f"[Watcher] New project directory detected: {event.src_path}")
+            with self.app.app_context():
+                serve_projects_from_filesystem(self.app, self.watch_path)
```

Ruff (0.8.2)

Undefined name `serve_projects_from_filesystem`

(F821)